### DISCUSSION TITLE
allow set engines in now.json, not only allow in package.json

### DISCUSSION THREAD

**Discussion by bluelovers** [vercel/vercel#4445] [Ideas] [answered]
allow set engines in now.json, not only allow in package.json

because `engines` in `package.json`, will block when we are use other version of node.js in local

also if we deploy in multi services
will get conflict because they are not allow same value

**Comment by styfle** [comment, marked as answer]
Hi @bluelovers 

Thanks for the suggestion! We do not intend to add additional configuration to `now.json` (or `vercel.json`).

> engines in package.json, will block when we are use other version of node.js in local

This is the intended behavior because you wan to make sure development matches production. If you know that your app works with multiple versions of Node.js, you can specify a range.

See the npm docs here: https://docs.npmjs.com/files/package.json#engines

**Reply by bluelovers** [reply]
i want set engines, because default node version in now server is old at before

and heroku and now is not allow same value style
